### DISCUSSION TITLE
Failed to install Ruby 3.1.3 on Mac 13.0.1

### DISCUSSION THREAD

**Discussion by thomasvanholder** [rbenv/ruby-build#2099] [Build failures] [answered]
When trying to upgrade to the latest ruby version, the following error appears.

```bash
rbenv install 3.1.3
To follow progress, use 'tail -f /var/folders/4f/8l4w_pm54q111k4j50nnvsbr0000gn/T/ruby-build.20221129100347.17042.log' or pass --verbose
Downloading openssl-3.0.7.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/83049d042a260e696f62406ac5c08bf706fd84383f945cf21bd61e9ed95c396e
Installing openssl-3.0.7...
Installed openssl-3.0.7 to /Users/thomasvanholder/.rbenv/versions/3.1.3

Downloading ruby-3.1.3.tar.gz...
-> https://cache.ruby-lang.org/pub/ruby/3.1/ruby-3.1.3.tar.gz
Installing ruby-3.1.3...
ruby-build: using readline from homebrew
ruby-build: using gmp from homebrew

BUILD FAILED (macOS 13.0.1 using ruby-build 20221124)

Inspect or clean up the working tree at /var/folders/4f/8l4w_pm54q111k4j50nnvsbr0000gn/T/ruby-build.20221129100347.17042.0bb8Zs
Results logged to /var/folders/4f/8l4w_pm54q111k4j50nnvsbr0000gn/T/ruby-build.20221129100347.17042.log

....
make: *** [extract-gems-sequential] Error 1
```

When opening the file specified above, an more granular error related to fileutiles is listed.

```bash
Extracting bundled gem files...
/Users/thomasvanholder/.rbenv/versions/3.0.2/lib/ruby/3.0.0/rubygems/specification.rb:2235:in `check_version_conflict': can't activate fileutils-1.6.0, already activated fileutils-1.5.0 (Gem::LoadError)
```

Any ideas on how to solve this?

**Comment by mislav** [comment]
@hsbt Does Ruby needs `ruby` to build? Any idea why it would activate `~/.rbenv/versions/3.0.2` while building `3.1.3`?

**Reply by hsbt** [reply]
the Ruby binary is needed only `-dev` build. I'm not sure why this report mixed Ruby 3.1 and 3.0.

**Reply by mislav** [reply]
@hsbt Thank you for clarifying.

@thomasvanholder I think you should share more from your log with us in order to track this down. Pasting a log as a gist is a welcome approach.

**Reply by thomasvanholder** [reply]
@mislav, here is a [gist](https://gist.github.com/thomasvanholder/14492f63ed55a4dccf750c558dafa089#file-ruby-3-1-3-failed-build) with a full log

**Comment by JackGarrus** [comment]
I am having the exact same issue

**Reply by mislav** [reply]
That's not a lot to go on; sorry. You can help us by posting the full build log as a gist

**Comment by thomasvanholder** [comment]
Duplicate of #2108

**Comment by thomasvanholder** [comment, marked as answer]
Uninstalling the older ruby version (3.0.2) and reinstalling the ruby (3.2.0) eventually worked for me.

```bash
rbenv uninstall 3.0.2
rbenv install 3.2.0
```
